interface/main/tabs/main.php

```diff
@@ -90,6 +90,18 @@ function isEncounterLocked( encounterId ) {
 	<?php } ?>
 }
 var webroot_url="<?php echo $web_root; ?>";
+
+// TODO: This needs to be in a <script src= ... file so that the big string doesn't make
+// the source file so wide. Not a huge deal, but would be nice.
+function xlj(key) {
+  var _translations=<?php require_once($GLOBALS['fileroot'] . "/interface/main/translations/javascript_translations_provider_controller.php"); ?>;
```

Comment:
This seems awfully convoluted and hard to follow.  Couldn't you just call generate_javascript_translation_mapping() right here?